src/create-support-ticket.ts

```diff
@@ -0,0 +1,201 @@
+import { SNSEvent } from "aws-lambda";
+import axios from "axios";
+import { getSecret } from "@aws-lambda-powertools/parameters/secrets";
+import {
+  SendMessageCommand,
+  SendMessageCommandOutput,
+  SendMessageRequest,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+import {
+  CreateTicketPayload,
+  HttpError,
+  SuspiciousActivityEvent,
+} from "./common/model";
+
+export const formatCommentBody = (event: SuspiciousActivityEvent): string => {
+  const htmlBody = [];
+  const eventDateTime = new Date(event.timestamp * 1000);
+
+  htmlBody.push(`<p><strong>Event Name</strong>: ${event.event_name}</p>`);
+
+  if (event.event_id) {
+    htmlBody.push(`<p><strong>Event ID</strong>: ${event.event_id}</p>`);
+  }
+
+  if (event.client_id) {
+    htmlBody.push(`<p><strong>Client ID</strong>: ${event.client_id}</p>`);
+  }
+
+  if (event.timestamp) {
+    htmlBody.push(
+      `<p><strong>Date and Time</strong>: ${eventDateTime.toUTCString()}</p>`
+    );
+  }
+
+  if (event.user?.user_id) {
+    htmlBody.push(`<p><strong>User ID</strong>: ${event.user.user_id}</p>`);
+  }
+
+  if (event.user?.session_id) {
+    htmlBody.push(
+      `<p><strong>Session ID</strong>: ${event.user.session_id}</p>`
+    );
+  }
+
+  if (event.user?.govuk_signin_journey_id) {
+    htmlBody.push(
+      `<p><strong>Gov UK Sign-in Journey ID</strong>: ${event.user.govuk_signin_journey_id}</p>`
+    );
+  }
+  return htmlBody.join("");
+};
+
+export const validateSuspiciousActivity = (
+  suspiciousActivityEvent: SuspiciousActivityEvent
+): void => {
+  if (
+    suspiciousActivityEvent.event_id === undefined ||
+    suspiciousActivityEvent.event_name === undefined ||
+    suspiciousActivityEvent.user?.session_id === undefined ||
+    suspiciousActivityEvent.user?.user_id === undefined ||
+    suspiciousActivityEvent.client_id === undefined ||
+    suspiciousActivityEvent.timestamp === undefined ||
+    suspiciousActivityEvent.reported_suspicious === undefined
+  ) {
+    throw new Error("Could not validate Suspicious Event Body");
+  }
+};
+
+export async function createTicket(
+  ticket: CreateTicketPayload,
+  apiUrl: string,
+  apiUsername: string,
+  apiToken: string
+): Promise<void> {
+  const token = `${apiUsername}:${apiToken}`;
+  const instance = axios.create({
+    baseURL: apiUrl,
+    headers: {
+      Authorization: `Basic ${Buffer.from(token).toString("base64")}`,
+      "Content-Type": "application/json",
+    },
+  });
+
+  try {
+    await instance.post("/tickets.json", ticket);
```

Comment:
Just wondering if it's better to declare URL's in a constants file

Reply:
Agreed, now moved to constant